### PR TITLE
Quake2 and Quetoo filter improvements

### PR DESCRIPTION
**Pull request by jdolan**
The GtkRadiant base filters are very Q3-centric. This patch aims to crutch up the FilterBrush function with logic that handles idTech2 `SURF_` flags like `SURF_TRANS33`, `SURF_TRANS66`, `SURF_HINT`, `SURF_SKIP`, etc. It also opens the door for other games to branch the base filters, should that be desired.

Additionally, this patch introduces a new filter for sky brushes, and this filter will work for idTech2 and idTech3 games.

### CODE REVIEW

File: radiant/filters.cpp [hunk 1]
```diff
@@ -259,6 +283,51 @@ bool FilterBrush( brush_t *pb ){
...
+							if ( f->texdef.flags & SURF_SKY ) {
+								filterbrush = true;
+							}
+							if ( strstr( f->pShader->getName(), "sky" ) ) {
+								filterbrush = true;
+							}
+							break;
+
+					}
+				}
```
Comment (jdolan) [inline radiant/filters.cpp:330]: This is the meat of the patch. It makes the stock GtkRadiant "exclude" filters work with Quake and Quake2 derivatives.

File: radiant/mainframe.cpp [hunk 2]
```diff
@@ -7808,6 +7824,20 @@ void MainFrame::OnFilterWorld(){
...
+		gtk_check_menu_item_set_active( GTK_CHECK_MENU_ITEM( item ), TRUE );
+	}
+	else{
+		gtk_check_menu_item_set_active( GTK_CHECK_MENU_ITEM( item ), FALSE );
+	}
+	g_bIgnoreCommands--;
+	PerformFiltering();
+	Sys_UpdateWindows( W_XY | W_CAMERA );
+}
+
```
Comment (jdolan) [inline radiant/mainframe.cpp:7840]: All of the changes to mainframe.cpp are to add the new sky filter to the menu, and to conditionalize some of the filter menus that do not pertain to idTech2, like patches. I'm not opposed to removing the conditionalizing if you find it offensive.

File: radiant/camwindow.cpp [hunk 3]
```diff
@@ -430,31 +430,75 @@ void CamWnd::Cam_MouseControl( float dtime ){
...
+		VectorClear( velocity );
+	}
+
+	// Save a screen update
+	if (!VectorCompare( vec3_origin, velocity ) || !VectorCompare( vec3_origin, avelocity ) ) {
+
+		// When m_bFreeMove is enabled, mousecontrol does the update
+		if ( m_bFreeMove ) {
+			return;
+		}
```
Comment (jdolan) [inline radiant/camwindow.cpp:500]: Hopefully I'm not getting too carried away here. But I've added some `noclip` style movement to the camera window that I think feels much nicer, and allows me to more easily select brushes and faces by moving / aiming more precisely.

File: radiant/mainframe.cpp [hunk 4]
```diff
@@ -830,9 +836,9 @@ static gint mainframe_keypress( GtkWidget* widget, GdkEventKey* event, gpointer
 static gint mainframe_keyrelease( GtkWidget* widget, GdkEventKey* event, gpointer data ){
 	unsigned int code = gdk_keyval_to_upper( event->keyval );
 
-	if ( gtk_accelerator_valid( event->keyval, (GdkModifierType)0 ) ) {
-		return TRUE;
-	}
+//	if ( gtk_accelerator_valid( event->keyval, (GdkModifierType)0 ) ) {
+//		return TRUE;
+//	}
```
Comment (jdolan) [inline radiant/mainframe.cpp:841]: I _think_ this was just a "performance optimization"? It needs to be commented out in order to send keyup events to non-special keys, which is required for the new smooth camera movement code. I did not notice any negative side effects by commenting this out.

**Comment by jdolan** [comment]
@TTimo Are you okay with me merging this PR?

**Comment by TTimo** [comment]
Sure, LGTM

**Pull request merged**